lib/url_for.ts

```diff
@@ -5,43 +5,73 @@ import prettyUrls from './pretty_urls';
 import Cache from './cache';
 const cache = new Cache<string>();
 
-function urlForHelper(path = '/', options) {
+/**
+ * url_for options type
+ * @example
+ * // to call this type
+ * type urlOpt = Parameters<typeof import('hexo-util')['url_for']>[1];
+ */
+interface url_for_options {
+  relative?: boolean;
+}
+
+/**
+ * get url relative to base URL (config_yml.url)
+ * @param path relative path inside `source` folder (config_yml.source_dir)
+ * @param options
+ * @returns
+ * @example
+ * // global `hexo` must be exist when used this function inside plugin
+ * const Hutil = require('hexo-util')
+ * console.log(Hutil.url_for.bind(hexo)('path/to/file/inside/source.css')); // https://example.com/path/to/file/inside/source.css
+ */
+function urlForHelper(path = '/', options: url_for_options | null = {}) {
   if (/^(#|\/\/|http(s)?:)/.test(path)) return path;
 
   const { config } = this;
 
-  options = Object.assign({
-    relative: config.relative_link
-  }, options);
+  options = Object.assign(
+    {
+      relative: config.relative_link
+    },
+    // fallback empty object when options filled with NULL
+    options || {}
+  );
 
   // Resolve relative url
   if (options.relative) {
     return relative_url(this.path, path);
   }
 
   const { root } = config;
-  const prettyUrlsOptions = Object.assign({
-    trailing_index: true,
-    trailing_html: true
-  }, config.pretty_urls);
+  const prettyUrlsOptions = Object.assign(
```

Comment:
You can keep the original code style here

Reply:
Looks like he used prettier. If eslint doesn't fail, I guess it doesn't matter.